### PR TITLE
docs: add examples for common usecases

### PR DESCRIPTION
**Pull request by dundargoc**

### CODE REVIEW

**Comment by stevengj** [comment]
Looks good to me!  Might be nice to also add an example of NFC and NFD normalization, e.g. transforming `"äöü"` between NFC and NFD:
* NFC: `"\u00e4\u00f6\u00fc"`
* NFD: `"a\u0308o\u0308u\u0308"` (uses [combining `" ̈"` U+0308](https://www.fileformat.info/info/unicode/char/0308/index.htm))

**Comment by dundargoc** [comment]
Done. Please take a look.

File: README.md [hunk 1]
```diff
@@ -86,3 +86,50 @@ the [utf8proc issues page on Github](https://github.com/JuliaLang/utf8proc/issue
...
+utf8proc_uint8_t *fold_str;
+utf8proc_map(str, 0, &fold_str, UTF8PROC_NULLTERM | UTF8PROC_CASEFOLD);
+printf("%s\n", fold_str);
+// ss
+free(fold_str);
+```
+
+### Normalization Form C/D (NFC/NFD)
+```c
+// Decompose `åäö` into `a\u0308o\u0308u\u0308`
```
Comment (stevengj) [inline README.md:126]:
```suggestion
// Decompose "\u00e4\u00f6\u00fc" = "äöü" into "a\u0308o\u0308u\u0308" (= "äöü" via combining char U+0308)
```

File: README.md [hunk 2]
```diff
@@ -86,3 +86,50 @@ the [utf8proc issues page on Github](https://github.com/JuliaLang/utf8proc/issue
...
+utf8proc_uint8_t input[] = {0xc3, 0xa4, 0xc3, 0xb6, 0xc3, 0xbc}; // åäö
```
Comment (stevengj) [inline README.md:127]:
```suggestion
utf8proc_uint8_t input[] = {0xc3, 0xa4, 0xc3, 0xb6, 0xc3, 0xbc}; // "\u00e4\u00f6\u00fc" = "äöü" in UTF-8
```

File: README.md [hunk 3]
```diff
@@ -86,3 +86,50 @@ the [utf8proc issues page on Github](https://github.com/JuliaLang/utf8proc/issue
...
+utf8proc_uint8_t *nfd= utf8proc_NFD(input);
```
Comment (stevengj) [inline README.md:128]:
```suggestion
utf8proc_uint8_t *nfd= utf8proc_NFD(input); // = {0x61, 0xcc, 0x88, 0x6f, 0xcc, 0x88, 0x75, 0xcc, 0x88}
```

**Comment by stevengj** [comment]
Thanks!

**Pull request merged**